compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -8517,14 +8519,32 @@ private STNode parseRightArrow() {
     }
 
     /**
-     * Parse parameterized type descriptor.
-     * parameterized-type-descriptor := map type-parameter | future type-parameter | typedesc type-parameter
+     * Parse map type descriptor.
+     * map-type-descriptor := `map` type-parameter
      *
      * @return Parsed node
      */
-    private STNode parseParameterizedTypeDescriptor(STNode parameterizedTypeKeyword) {
+    private STNode parseMapTypeDescriptor(STNode mapKeyword) {
         STNode typeParameter = parseTypeParameter();
-        return STNodeFactory.createParameterizedTypeDescriptorNode(parameterizedTypeKeyword, typeParameter);
+        return STNodeFactory.createMapTypeDescriptorNode(mapKeyword, typeParameter);
+    }
+
+    /**
+     * Parse future type descriptor.
+     * future-type-descriptor := `future` [type-parameter]
+     *
+     * @return Parsed node
+     */
+    private STNode parseFutureTypeDescriptor(STNode futureKeywordToken) {
```

Comment:
Can we use a single function to parse all parameterized types?. We can get the SyntaxKind from parsed keyword